rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinTestUtils.java

```diff
@@ -48,4 +55,32 @@ public static Sha256Hash createHash(int nHash) {
 
         return Sha256Hash.wrap(bytes);
     }
+
+    public static List<BtcECKey.ECDSASignature> extractSignaturesFromTxInput(TransactionInput txInput) {
+        Script scriptSig = txInput.getScriptSig();
+        List<ScriptChunk> chunks = scriptSig.getChunks();
+        Script redeemScript = new Script(chunks.get(chunks.size() - 1).data);
+        RedeemScriptParser parser = RedeemScriptParserFactory.get(redeemScript.getChunks());
+
+        // This chunk may be 0 or 1 depending on if the federation is spending funds from standard multisig or erp
+        int flowOpCodeChunk;
+        RedeemScriptParser.MultiSigType multiSigType = parser.getMultiSigType();
+
+        if (multiSigType == RedeemScriptParser.MultiSigType.STANDARD_MULTISIG
+                || multiSigType == RedeemScriptParser.MultiSigType.FAST_BRIDGE_MULTISIG
+        ) {
+            flowOpCodeChunk = chunks.size() - 1;
+        } else {
+            flowOpCodeChunk = chunks.size() - 2;
+        }
+
+        List<BtcECKey.ECDSASignature> signatures = new ArrayList<>();
+        for (int i = 1; i < flowOpCodeChunk; i++) {
+            ScriptChunk chunk = chunks.get(i);
+            if (chunk.isOpCode() || chunk.data.length > 0) {
```

Comment:
I just recreated `countMissingSignatures` method. See [co.rsk.peg.BridgeUtils#countMissingSignatures](https://github.com/rsksmart/rskj/blob/master/rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java#L570).

Before the condition was:

```
if (!chunk.isOpCode() && chunk.data.length == 0) {
                unsigned++;
            }
```

Since Kelvin left a comment asking to have the condition as positive, I changed it to how it is now.

Reply:
I don't think those 2 are equivalent. Did you try doing a truth table?

Also careful with copy pasting, what is the content of a chunk with a signature? Should it be an op code?

Reply:
Same here.

Reply:
Done